examples/staking-broker/src/wasm.rs

```diff
@@ -0,0 +1,266 @@
+// This file is part of Gear.
+//
+// Copyright (C) 2024 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+//
+// This is a very basic implementation of a staking broker program for demo purpose only.
+// It accepts users messages and bond, unbond or nominate some validators on their behalf.
+// It is not concerned with some important constraints like unbonding period etc., so a real
+// implementation of a liquid staking contract, for example, would be more complex.
+
+use gbuiltin_staking::*;
+use gstd::{debug, msg, prelude::*, ActorId};
+use hashbrown::HashMap;
+use hex_literal::hex;
+use parity_scale_codec::Encode;
+
+// Staking proxy builtin actor program id (hardcoded for all runtimes)
+const BUILTIN_ADDRESS: ActorId = ActorId::new(hex!(
+    "77f65ef190e11bfecb8fc8970fd3749e94bed66a23ec2f7a3623e785d0816761"
+));
+
+// Runtime AccountId type
+pub type AccountId = [u8; 32];
```

Comment:
this can use just ActorId type since their encoding matches

Reply:
moreover a few lines below you use both ActorId and this AccountId :/

Reply:
These are conceptually two different things, aren't they - a program address in actor space and an ordinary account id?
However, you're right that this `AccountId` declaration is redundant: we should reuse the `AccountId` from the `gbuiltin_staking` crate - the type that is used in API functions declarations. Removed it.